pandas/tests/sparse/test_array.py

```diff
@@ -414,27 +460,33 @@ def test_set_fill_value(self):
         arr.fill_value = 2
         assert arr.fill_value == 2
 
+        # XXX: this seems fine? You can construct an integer
```

Comment:
As said before, I personally like more the behaviour as it was before. 
Was there a specific use case to allow this flexibility?